moshi/src/main/java/com/squareup/moshi/Types.java

```diff
@@ -386,6 +386,23 @@ static Type resolveTypeVariable(Type context, Class<?> contextRawType, TypeVaria
     return unknown;
   }
 
+  /**
+   * Returns true if this is a boxed primitive type like {@code Integer}. Returns false for all
+   * other types types including unboxed primitives and {@code java.lang.Void}.
+   */
+  static boolean isBoxedPrimitive(Type type) {
+    return type.equals(Boolean.class)
+        || type.equals(Byte.class)
+        || type.equals(Character.class)
+        || type.equals(Double.class)
+        || type.equals(Float.class)
+        || type.equals(Integer.class)
+        || type.equals(Long.class)
+        || type.equals(Short.class)
+        || type.equals(String.class)
+        || type.equals(Object.class);
```

Comment:
I renamed from `isBoxedPrimitive` to `isAllowedPlatformType`. Not super confident on that name.